src/Spain/Mews.Fiscalizations.Spain/Nif/NifValidator.cs

```diff
@@ -17,11 +18,14 @@ public NifValidator(X509Certificate certificate, TimeSpan httpTimeout)
         }
         private SoapClient SoapClient { get; }
 
-        public async Task<Response> CheckNif(Request model)
+        public async Task<ResponseResult<Response>> CheckNif(Request model)
         {
             var request = Convert(model);
             var response = await SoapClient.SendAsync<Entrada, Salida>(request);
-            return Convert(request, response);
+            return response.IsSuccess.Match(
+                t => new ResponseResult<Response>(successResult: Convert(request, response.SuccessResult)),
+                f => new ResponseResult<Response>(errorResult: response.ErrorResult)
```

Comment:
I would prefer static methods over invocations with named optional parameters. I'd find that nicer.

But more importantly, ResponseResult now allows to be created with both a success result and an error result at the same time. That should not be possible.